### PR TITLE
Disable pauses for auto pause enabled updates

### PR DESCRIPTION
**Pull request by zorro786**
Add IT

Signed-off-by: Abdul Qadeer <aqadeer@paypal.com>

### Description:
This PR prevents pauses on auto pause enabled job updates.

### Testing Done:
Ran UTs - all passed
E2E tests (failures unrelated to this commit)

### CODE REVIEW

File: src/main/java/org/apache/aurora/scheduler/updater/JobUpdateControllerImpl.java [hunk 1]
```diff
@@ -255,6 +255,16 @@ public void assertNotUpdating(IJobKey job) throws JobUpdatingException {
...
   public void pause(final IJobUpdateKey key, AuditData auditData) throws UpdateStateException {
     requireNonNull(key);
+    Optional<IJobUpdateDetails> update = storage.read(p -> p.getJobUpdateStore()
+            .fetchJobUpdate(key));
+    Boolean isAutoPauseEnabled = update.isPresent() && isAutoPauseEnabled(update.get()
+            .getUpdate()
+            .getInstructions()
+            .getSettings()
+            .getUpdateStrategy());
+    if (isAutoPauseEnabled && instancesSeen.containsKey(key)) {
```
Comment (ridv) [inline src/main/java/org/apache/aurora/scheduler/updater/JobUpdateControllerImpl.java:265]: If we check if the instancesSeen contains the key, we can know the update started allowing us to pause for a brief time before the update starts if we do this check.  It's probably overkill to do that in this case. I think in this case we can just check to see if it's auto pause enabled by itself.

**Review by ridv — COMMENTED** [review]
LGTM minus one minor nit

File: src/main/java/org/apache/aurora/scheduler/updater/JobUpdateControllerImpl.java (hunk 1 above)
Comment (zorro786) [reply src/main/java/org/apache/aurora/scheduler/updater/JobUpdateControllerImpl.java]: Yeah on second look it seems redundant. Fixed.

**Review by ridv — APPROVED** [review]
:shipit:

**Pull request merged**